File: OneQuestionPerDay/83/remove_duplicates_from_sorted_list.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func deleteDuplicates(head *ListNode) *ListNode {
	if head == nil {
		return head
	}
	elements := map[int]bool{head.Val: true}
	prev, cur := head, head.Next
	for cur != nil {
		if elements[cur.Val] {
			prev.Next, cur = cur.Next, cur.Next
		} else {
			elements[cur.Val] = true
			prev, cur = cur, cur.Next
		}
	}
	return head
}

File: OneQuestionPerDay/83/remove_duplicates_from_sorted_list.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if head is None:
            return head
        elments = {head.val}
        prev, cur = head, head.next
        while cur is not None:
            if cur.val not in elments:
                elments.add(cur.val)
                prev, cur = cur, cur.next
            else:
                prev.next = cur.next
                cur = cur.next
        return head
